### ISSUE TITLE
Cuda OOM errors, even on A100 with batch size 1

### ISSUE THREAD

**Issue by Tylersuard** [RuslanKhalitov/ChordMixer#6]
Hello.  I am getting Cuda OOM errors at around 40% of the first epoch when training on the Sus_Bos problem.  Is there any way to scale the model and/or problem down further so I can train on Colab?

**Comment by RuslanKhalitov** [comment]
The Sus_Bos task is the heaviest one among the Genbank datasets. I do confirm that with the current config, the OOM error appears when training on an A100 with 40Gb. If you wanna run the experiment on Collab, you can change the config values (track_size=8; hidden_size=96) to make ChordMixer blocks lighter. I checked that the training runs without errors on V100/32Gb but slightly downgrades the accuracy. For practical reasons, I changed the [default values](https://github.com/RuslanKhalitov/ChordMixer/blob/e15ebc4b5ee1ceeae45013353363e812f400fe9a/conf/config.yaml#L76) to those that fit a standard 32Gb Gpu.

By the way, I saw your Colab notebook. Good job! I noticed you changed the batch size in the config directly; however, the codebase does not support training with bs=1. You still train using the bs=2 because of [this line](https://github.com/RuslanKhalitov/ChordMixer/blob/9c66ee1f7e8cc536544a206125a8c37614adca37/dataloaders/utils/bucket_sampler.py#LL23C45-L23C45) in the BatchSampler. So I recommend keeping batch size=2 for these experiments.

**Comment by Tylersuard** [comment]
@RuslanKhalitov Thank you so much, this is really helpful!  I appreciate you taking the time to reply.

**Comment by RuslanKhalitov** [comment]
Thanks for checking this out. Btw, I have finished the training on V100/32Gb.
Here is the [report](https://api.wandb.ai/links/deepntnu/x87rn8ol)
Your metrics and the training should have similar behavior.

**Comment by Tylersuard** [comment]
Thank you again.  I was able to get it up and running for me on Colab.  I found a line in the model file that sets the number of blocks based on the log of the maximum sequence length.  Once I changed that code to manually set the number of blocks, everything worked perfectly for me.

I was able to create an audio classifier that takes audio as a sequence of samples, rather than as a spectrogram (which is common right now due to audio files containing many samples).  I was able to get a 77% accuracy on 3 classes, which is great!  I will keep working on improving that accuracy.  Thank you for your help and for releasing this repo.
